src/ripple/consensus/Validations.h

```diff
@@ -580,17 +582,17 @@ class Validations
             // Check that validation sequence is greater than any non-expired
             // validations sequence from that validator
             auto const now = byLedger_.clock().now();
-            SeqEnforcer<Seq>& enforcer = seqEnforcers_[key];
+            SeqEnforcer<Seq>& enforcer = seqEnforcers_[nodeID];
             if (!enforcer(now, val.seq(), parms_))
                 return ValStatus::badSeq;
 
             // This validation is a repeat if we already have
-            // one with the same id for this key
-            auto const ret = byLedger_[val.ledgerID()].emplace(key, val);
+            // one with the same id and signing key for this node
```

Comment:
I think `repeatedID` actually should have been removed as part of the  prefer by branch work (which you suggested https://github.com/ripple/rippled/pull/2300#discussion_r164039567). If a node issues two validations with the same ID (using either the same or different signing keys), the seq enforcer above will ignore all but the first.

Agree?

Reply:
How about (the unexpected case) when there's multiple validations with the same ledger id from the same validator but with different sequences?
We could just treat it normally (add the new validation assuming it has a higher sequence) and not need `repeatedID`.
I was mainly asking because I don't think this change to the comment is accurate.

Reply:
I believe the ledger sequence number is part of what is hashed to create the ledger id, so you can't really have two validations for the same id but different sequence numbers.

I agree that the comment is now inaccurate.

Reply:
Right, it would be a malformed validation, but that doesn't mean someone won't send one.

Reply:
Good point.  

I like 
> We could just treat it normally (add the new validation assuming it has a higher sequence) and not need repeatedID.

I don't see `repeatedID` really giving us much useful information; the increasing sequence number invariant is the essential one.

Reply:
Actually, what would happen if a trusted validator only sends validations for a single id but with incrementing sequence numbers?
It seems like it would be as if they only validated that single id, but that validation never expires/is always current (assuming they keep resigning and updated the sign time). If we ever stopped keeping that ledger (via online delete?) we would keep re-acquiring it in order to update the ledger trie.
Would any of that be an issue?

Reply:
If we were unable to acquire it, that validator's validation would not be added to the trie and effectively be ignored. Impact wise, I don't see this as different than a validator sending validations for random ledger ids that we can't acquire.

Reply:
I'm wondering about if it's a valid ledger id that we can acquire.
If we haven't acquired it yet, and we have a prior validation with the same id but lower seq, we would cancel and then restart the acquire attempt.
If/once we have acquired the ledger, we would have this entry in the trie with the real seq that persists until the bad validator stops sending the validations.